### PR TITLE
Fix #5613 and #6137: network flashing when booting status changes, and level emitter cycle restart (other approach)

### LINKED ISSUE #5613: Ad hoc network flashing
### Describe the bug

https://streamable.com/ppn36n

The left ad hoc network shouldn't be flashing like that.

### How to reproduce the bug

Place a line of cables with a creative cell and multiple terminals. Break a cable between some terminals, and watch as the terminals that remain powered flash for 1 tick.

### Expected behavior

No flashing.

### Additional details

_No response_

### Which minecraft version are you using?

1.17

### On which mod loaders does it happen?

Fabric

### Crash log

n/a

### PR DESCRIPTION
**Pull request by Technici4n**
I changed when the channels are finalized (this is what sends the channel change events to grid node listeners) to be after the grid has finished booting so `isActive` checks properly succeed in there. There are two ways to filter out unwanted notifications while the grid is booting:
- Check `state != State.GRID_BOOTED`: this will only be called if either power or channel state changes. Checking `isActive` there will properly return the new active state. This is appropriate when activeness is only used for display purposes or decided whether some logic should be skipped.
- Check `getMainNode().hasGridBooted()`: this will be called if either the power or channel state changes, and also every time the grid finished booting. This is appropriate when the device was sleeping while booting and might need to be awakened. For example, for ME interfaces, this has to be used to avoid issues such as #5603.

### CODE REVIEW

File: src/main/java/appeng/parts/storagebus/StorageBusPart.java [hunk 1]
```diff
@@ -141,11 +141,13 @@ public StorageBusPart(IPartItem<?> partItem) {
...
-        if (this.wasActive != currentActive) {
-            this.wasActive = currentActive;
-            this.getHost().markForUpdate();
-            remountStorage();
+        if (reason != IGridNodeListener.State.GRID_BOOT) {
+            var currentActive = this.getMainNode().isActive();
+            if (this.wasActive != currentActive) {
+                this.wasActive = currentActive;
+                this.getHost().markForUpdate();
+                remountStorage();
```
Comment (Technici4n) [inline src/main/java/appeng/parts/storagebus/StorageBusPart.java:149]: Should storage be unmounted when booting status changes?

Comment (shartte) [reply src/main/java/appeng/parts/storagebus/StorageBusPart.java:149]: If we unmount storage, would we also unmount drives?

Comment (Technici4n) [reply src/main/java/appeng/parts/storagebus/StorageBusPart.java:149]: Yeah, I think so.

Comment (Technici4n) [reply src/main/java/appeng/parts/storagebus/StorageBusPart.java:149]: It would probably cause unwanted level emitter updates though, wouldn't it?

**Comment by shartte** [comment]
Yeah we should not do that I think

Technici4n ***@***.***> schrieb am Fr., 22. Apr. 2022, 13:44:

**Pull request merged**